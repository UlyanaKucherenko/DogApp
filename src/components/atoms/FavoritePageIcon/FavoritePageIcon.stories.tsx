import type { Meta, StoryObj } from '@storybook/react';
import { FavoritePageIcon } from '.';

const meta = {
  title: 'Atoms/FavoritePageIcon',
  component: FavoritePageIcon,
  argTypes: {
    state: {
      options: ['active', 'default'],
      control: 'radio'
    }
  }
} satisfies Meta<typeof FavoritePageIcon>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    state: 'default'
  }
};
