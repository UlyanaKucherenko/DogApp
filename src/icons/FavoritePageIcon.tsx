import { SvgIcon } from '@mui/material';

import FavoritePageDefaultIcon from '../assets/icons/FavoritePageDefaultIcon.svg?react';
import FavoritePageActiveIcon from '../assets/icons/FavoritePageActiveIcon.svg?react';

type FavoritePageIconProps = {
  state: 'active' | 'default';
};

export const FavoritePageIcon = ({
  state = 'default'
}: FavoritePageIconProps) => {
  switch (state) {
    case 'active': {
      return <SvgIcon component={FavoritePageActiveIcon} inheritViewBox />;
    }
    default: {
      return <SvgIcon component={FavoritePageDefaultIcon} inheritViewBox />;
    }
  }
};
