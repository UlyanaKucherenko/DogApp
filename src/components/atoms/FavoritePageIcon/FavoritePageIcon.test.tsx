import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react';
import * as stories from './FavoritePageIcon.stories';

describe('FavoritePageIcon', () => {
  const { Primary } = composeStories(stories);
  it('should match snapshot Primary', () => {
    const { container } = render(<Primary />);
    expect(container).toMatchSnapshot();
  });
});
