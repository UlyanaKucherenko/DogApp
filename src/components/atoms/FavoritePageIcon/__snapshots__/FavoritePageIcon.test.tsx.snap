// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritePageIcon should match snapshot Primary 1`] = `
<div>
  <test-file-stub
    aria-hidden="true"
    classname="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    focusable="false"
  />
</div>
`;
